### PR TITLE
feat: Add proxy to build script for device-usb-service

### PR DESCRIPTION
**Pull request by seanohair22**
**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [x] I have opened a PR for the related docs change (if not, why?)
[  Docs PR for proxy](https://github.com/edgexfoundry/edgex-docs/pull/1184)

## Testing Instructions

Add proxy to docker (ping me for specific proxy setup instructions)
Test onsite or on VPN
Run `make docker`, the service should build.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,6 +33,9 @@ docker:
 	docker build . \
 		--build-arg ADD_BUILD_TAGS=$(ADD_BUILD_TAGS) \
 		--label "git_sha=$(GIT_SHA)" \
+		--build-arg http_proxy=$(HTTP_PROXY) \
+		--build-arg https_proxy=$(HTTPS_PROXY) \
+		--build-arg no_proxy=$(NO_PROXY) \
```
Comment (ajcasagrande) [inline Makefile:38]: Should these use the lowercase form as well? aka `--build-arg http_proxy=$(http_proxy)`, which i think can just be shortened to `--build-arg http_proxy`

**Review by presatish — COMMENTED** [review]
The changes look fine but I am not able to test as it looks like vpn certificate for my linux machine has been expired or has some issues because of which I am unable to test.

File: Makefile (hunk 1 above)
Comment (seanohair22) [reply Makefile:38]: I followed what was present in the device-onvif-service, that way the environment variables only need to be set with one format for building both services...

Comment (ajcasagrande) [reply Makefile:38]: ok, that seems fair, as long as the documentation matches up, which I believe it does when I reviewed that last week.

**Review by ajcasagrande — APPROVED** [review]

**Pull request merged**